### PR TITLE
interfaces: add microceph interface

### PR DESCRIPTION
**Pull request by stgraber**
This interface makes it possible to control MicroCeph.

It will be used by another snap, MicroCloud to bring up MicroCeph and LXD clusters automatically.

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

**Review by mardy — COMMENTED** [review]
I wonder if we shouldn't use a more traditional plug-slot approach, with a slot for the provider of the socket and an attribute that allows retrieving the path. Then it could be a generic interface, instead of a microceph-specific one.

We already have posix-mq and shared-memory interfaces, and we have a request to add a way to have XDP sockets work. I wonder if we should have a single interface for socket operations, with different attributes depending on the socket type. But then different types require different privileges (and therefore levels of trust), so maybe this is not feasible.

Still, a netlink-socket interface might make sense?

File: interfaces/builtin/microceph.go
Comment (stgraber) [reply interfaces/builtin/microceph.go]: Fixed

**Comment by stgraber** [comment]
@mardy a more generic way to handle such sockets would certainly be nice, however as those are pretty privileged interfaces, we need something that the store can easily detect and allow for manual review so I don't know that this one would be a prime candidate for that?

**Review by Meulengracht — COMMENTED** [review]
The changes themselves look fine to me - however I can't say anything as to if it should be merged or not.

File: interfaces/builtin/microceph.go
Comment (mvo5) [inline interfaces/builtin/microceph.go:45]: Our most recent interfaces are called `-support` when it comes to controlling things (e.g. `cups-control, snapd-control, ...`. However the `lxd` inteface breaks this pattern. I think `microceph` control would be slightly better but @pedronis has the final say in this.

**Review by mvo5 — APPROVED** [review]
This looks fine, needs a security review and final yes/no from Samuele

File: interfaces/builtin/microceph.go
Comment (stgraber) [reply interfaces/builtin/microceph.go:45]: Ah yeah, no issue with `microceph-control` if that's the current pattern.

I based this on both `docker` and `lxd` which are indeed older and likely predate `-control`.

**Review by alexmurray — APPROVED** [review]
In general this LGTM from a security point-of-view, although it does feel a bit strange to be encoding the path of a snap specific socket into snapd itself - this feels more what a content interface should provide IMO.

**Review by pedronis — APPROVED** [review]
lgtm

File: interfaces/builtin/microceph.go
Comment (stgraber) [reply interfaces/builtin/microceph.go:45]: That socket will always give full control, so no plans for a read-only version.

**Pull request merged**
